nueljoe/healthcare
# feature(courses): course modules and lessons

#### What does this PR do?
Implement course module and lecture management

#### Description of Task to be completed?
- implement the `POST /courses/:slug/modules` endpoint for creating a module in a course

- implement the `GET /courses/:slug/modules` endpoint to fetch all the modules of a course

- implement the `PATCH /courses/:slug/modules/:moduleId` endpoint to update the details of a module

- implement the `PATCH /courses/:slug/modules/:moduleId/status` endpoint to publish/unpublish a module

- implement the `PATCH /courses/:slug/modules/:moduleId/position` endpoint to change the position of a module within a course

- implement the `DELETE /courses/:slug/modules/:moduleId` endpoint to delete a module

- implement the `POST /courses/:slug/modules/:moduleId/lectures` endpoint to created a new lecture in a module of a course
    
- implement the `GET /courses/:slug/modules/:moduleId/lectures` endpoint to fetch the lectures in a module of a course
    
- implement the `PATCH /courses/:slug/lectures/:lectureId` endpoint to update the content of a lecture
    
- implement the `GET /courses/:slug/lectures/:lectureId` endpoint to fetch a lecture
    
- implement the `DELETE /courses/:slug/lectures/:lectureId` endpoint to delete a lecture

- implement the `PATCH /courses/:slug/lectures/:lectureId/status` endpoint to publish/unpublish a course

- implement the `PATCH /courses/:slug/lectures/:lectureId/position` endpoint to move a course within a module or between modules

#### How should this be manually tested?
- clone the repository
- cd into the project directory and run `cd api` to CD into the `api` directory
- see the README.md file in the `api` directory for the preliminary steps
- ensure that a mysql database in installed and running on your device
- see the postman documentation on `Course Modules` and `Course Lectures`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
N/A

#### Screenshots (if appropriate)
N/A
#### Questions:
